### DISCUSSION TITLE
wire:current is not working when the component re-hydrated

### DISCUSSION THREAD

**Discussion by brunocbatista** [livewire/livewire#9074] [7. Bugs (archived)]
### Livewire version

v3.5.17

### Laravel version

v11.34.2

### Which PHP version are you using?

PHP 8.4

### What is the problem?

When we have a component with wire:current and it is re-hydrated, the component loses its style. For example, we have a list of users with an InfinityScroll, the selected user should be displayed with a different style, but the style is lost when the component is re-hydrated.

### Code snippets

```blade
<div>
    @foreach($users as $user)
        <a href="{{ route('users.show', ['user' => $user->id]) }}" class="no-underline text-black" wire:current="text-red-500">
            <p>{{ $user->name }}</p>
            <p>{{ $user->email }}</p>
        </a>
    @endforeach

    <div x-intersect="$wire.call('loadMore')"></div>
</div>
```

```php
<?php

namespace App\Livewire\Users;

use App\Models\User;
use Illuminate\View\View;
use Livewire\Component;

class Show extends Component
{
    public ?User $user;

    public function render(): View
    {
        return view('livewire.users.show');
    }
}
```

### Wirebox URL

not possible in wirebox

### How do you expect it to work?

I think the rehydration method can check if the component has the wire:current attribute and add the style to the active link.

### Please confirm (incomplete submissions will not be addressed)

- [X] I have provided easy and step-by-step instructions to reproduce the bug.
- [X] I have provided code samples as text and NOT images.
- [X] I understand my bug report will be removed if I haven't met the criteria above.

**Comment by joshhanley** [comment]
@brunocbatista Thank you for the report, however we have been unable to reproduce this given the information available. Please submit a code snippet we can copy and paste into our own apps with steps to reproduce. Be sure to include any Blade variable definitions that are used and use a little code as possible to reproduce the issue. Then, we will be happy to help.

**Reply by danie-ramdhani** [reply]
simple reproduction:

```blade
<button type="button" wire:click="$refresh">
    Refresh
</button>

<div>
    {{-- didn't work with alpine binding --}}
    <a x-bind:href="window.location.pathname" wire:current="text-red-500" x-text="window.location.pathname"></a>
    {{-- didn't work after livewire morph the page --}}
    <a href="/foo" wire:current="text-red-500">foo</a>

    {{-- workaround --}}
    <a href="/foo" wire:replace.self wire:current="text-red-500">foo</a>
    <a href="/foo" wire:ignore.self wire:current="text-red-500">foo</a>
    <a x-bind:href="window.location.pathname"
   x-bind:class="{
       'text-red-500': new URL($el.href, window.location.origin).pathname === window.location.pathname
   }"
   x-text="window.location.pathname"></a>
</div>
```

**Reply by ghabriel25** [reply]
@joshhanley This issue still exists on latest version. Adding `wire:ignore.self` solved the problem atm. If this is not an issue/bug, could I add some context on the docs?

**Reply by lrljoe** [reply]
@ghabriel25 

> This issue still exists on latest version. Adding `wire:ignore.self` solved the problem atm. If this is not an issue/bug, could I add some context on the docs?

Providing a "not working" example would help!  The example originally provided is incomplete.

**Reply by joshhanley** [reply]
@ghabriel25 never said it wasn't an issue or a bug 🙂 It should be fixed, rather than adding a note to the docs. PRs welcome!

**Reply by ghabriel25** [reply]
@lrljoe @joshhanley I'll provide wirebox example later. Thanks for the response

**Comment by ghabriel25** [comment]
@joshhanley @lrljoe Here is the wirebox example: https://wirebox.app/b/yr5r3.

On `settings` page, there is a form with input. Just make the validation fails then you can see `wire:current` not working.

**Reply by joshhanley** [reply]
@ghabriel25 your wirebox is private

**Reply by ghabriel25** [reply]
@joshhanley Sorry, I just notice that

**Reply by joshhanley** [reply]
@ghabriel25 thanks! I've submitted a PR #9384 with a fix
